DRIQuery.cpp

```diff
@@ -37,6 +40,20 @@ std::list<DriverConfiguration> DRIQuery::queryDriverConfigurationOptions(const G
         DriverConfiguration config;
         config.setScreen(i);
 
+
+        char *p = (char *)(*(this->getGlExtensionsString))(GL_EXTENSIONS);
+        if (p == nullptr) std::cout << "No GL extensions available" << std::endl;
+        //else std::cout << "GL extensions: " << p << std::endl;
+
+        /* Check if driver has mesa query extension or not? */
+        const char *extensionString;
+        extensionString = (*(this->getGlxExtensionsString))(display, i);
+        std::string possibleExts (extensionString);
+        if (possibleExts.find("GLX_MESA_query_renderer") == std::string::npos) {
+            std::cerr << "Closed source driver!!" << std::endl;
+            return std::move(configurations);
```

Comment:
You are just returning the configuration here.
How exactly you will inform the end-user that the closed source driver is not supported by adriconf?

Reply:
in L53 a simple error message will be shown in cmd line and remaining same as the case when you can't open the disaplay. But, if you have any other ideas like to pop up a new error message window, then we need to do work in GUI.cpp class.

Reply:
Well, my idea was giving some kind of visual feedback to the user, so making some changes to the gui class would make a lot of sense for me. Or even checking this kind of things before starting the gui at all, ie: check and give a dialog before creating a gui.cpp object.
I don't think people will check the output of a gui application. At least this can avoid some fake bug reports.